server/src/test/java/org/elasticsearch/search/aggregations/pipeline/DateDerivativeIT.java

```diff
@@ -262,27 +274,43 @@ public void testSingleValuedFieldNormalised_timeZone_CET_DstEnd() throws Excepti
         List<? extends Bucket> buckets = deriv.getBuckets();
         assertThat(buckets.size(), equalTo(4));
 
-        assertBucket(buckets.get(0), new DateTime("2012-10-27", timezone).toDateTime(DateTimeZone.UTC), 1L, nullValue(), null, null);
-        assertBucket(buckets.get(1), new DateTime("2012-10-28", timezone).toDateTime(DateTimeZone.UTC), 2L, notNullValue(), 1d, 1d / 24d);
+        DateFormatter dateFormatter = DateFormatter.forPattern("yyyy-MM-dd").withZone(ZoneOffset.UTC);
+
+        ZonedDateTime expectedKeyFirstBucket =
+            LocalDate.from(dateFormatter.parse("2012-10-27")).atStartOfDay(timezone).withZoneSameInstant(ZoneOffset.UTC);
+        assertBucket(buckets.get(0), expectedKeyFirstBucket, 1L, nullValue(), null, null);
+
+        ZonedDateTime expectedKeySecondBucket =
+            LocalDate.from(dateFormatter.parse("2012-10-28")).atStartOfDay(timezone).withZoneSameInstant(ZoneOffset.UTC);
+        assertBucket(buckets.get(1), expectedKeySecondBucket, 2L, notNullValue(), 1d, 1d / 24d);
+
         // the following is normalized using a 25h bucket width
-        assertBucket(buckets.get(2), new DateTime("2012-10-29", timezone).toDateTime(DateTimeZone.UTC), 3L, notNullValue(), 1d, 1d / 25d);
-        assertBucket(buckets.get(3), new DateTime("2012-10-30", timezone).toDateTime(DateTimeZone.UTC), 4L, notNullValue(), 1d, 1d / 24d);
+        ZonedDateTime expectedKeyThirdBucket =
+            LocalDate.from(dateFormatter.parse("2012-10-29")).atStartOfDay(timezone).withZoneSameInstant(ZoneOffset.UTC);
+        assertBucket(buckets.get(2), expectedKeyThirdBucket, 3L, notNullValue(), 1d, 1d / 25d);
+
+        ZonedDateTime expectedKeyFourthBucket =
+            LocalDate.from(dateFormatter.parse("2012-10-30")).atStartOfDay(timezone).withZoneSameInstant(ZoneOffset.UTC);
+        assertBucket(buckets.get(3), expectedKeyFourthBucket, 4L, notNullValue(), 1d, 1d / 24d);
     }
 
     /**
      * also check for time zone shifts that are not one hour, e.g.
      * "Asia/Kathmandu, 1 Jan 1986 - Time Zone Change (IST → NPT), at 00:00:00 clocks were turned forward 00:15 minutes
      */
+    // This test fails because we cannot parse negative epoch milli seconds yet... but perhaps we dont have to if we use instants in the
```

Comment:
Is this a blocker for merging? Do we need to have a follow-up discussion about negative epoch millis then? I was under the impression that they will not be supported.

Reply:
this is a leftover